### ISSUE TITLE
AST serialization in Scala

### ISSUE THREAD

**Issue by mwu-tow** [luna/luna#792]
### Summary
Scala AST should be serializible to JSON.

### Value
Being able to represent AST as JSON allows passing it through language boundaries without need for complex interfaces. AST can be just passed as a String.

### Specification
* circe is the preferred library to deal with serialization 
* all AST types / traits that Parser generates must be possible to serialize to JSON
* parts of AST not generated by Parser (like documented AST or nodes appearing only after resolving macros) are not required to be supported
* wherever possible we should use automatic derivation
* JSON deserialization in Scala is not part of this task

### Acceptance Criteria & Test Cases
For any AST node value `ast.asJson` (or equivalent) should compile and yield JSON.
